web/ui/react-app/src/pages/Status.tsx

```diff
@@ -1,6 +1,94 @@
-import React, { FC } from 'react';
+import React, { FC, Fragment } from 'react';
 import { RouteComponentProps } from '@reach/router';
+import { Table, Alert, Row } from 'reactstrap';
+import useFetch from './useFetch';
 
-const Status: FC<RouteComponentProps> = () => <div>Status page</div>;
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
 
-export default Status;
+const ENDPOINTS = ['../api/v1/status/runtimeinfo', '../api/v1/status/buildinfo', '../api/v1/alertmanagers'];
+const sectionTitles = ['Runtime Information', 'Build Information', 'Alertmanagers'];
+
+interface StatusConfig {
+  [k: string]: { title: string; normalizeValue?: (v: any) => any };
+}
+
+type StatusPageState = Array<{ [k: string]: string }>;
+
+const normalizeAlertmanagerValue = (alertMgrs: { url: string }[]) => {
+  return alertMgrs.map(({ url }) => (
+    <a key={url} className="mr-5" href={url}>
+      {url}
+    </a>
+  ));
+};
+
+export const statusConfig: StatusConfig = {
+  StartTime: { title: 'Start time', normalizeValue: (v: string) => new Date(v).toUTCString() },
+  CWD: { title: 'Working directory' },
+  ReloadConfigSuccess: {
+    title: 'Configuration reload',
+    normalizeValue: (v: boolean) => (v ? 'Successful' : 'Unsuccessful'),
+  },
+  LastConfigTime: { title: 'Last successful configuration reload' },
+  ChunkCount: { title: 'Head chunks' },
+  TimeSeriesCount: { title: 'Head time series' },
+  CorruptionCount: { title: 'WAL corruptions' },
+  GoroutineCount: { title: 'Goroutines' },
+  StorageRetention: { title: 'Storage retention' },
+  activeAlertmanagers: {
+    title: 'Active',
+    normalizeValue: normalizeAlertmanagerValue,
+  },
+  droppedAlertmanagers: {
+    title: 'Dropped',
+    normalizeValue: normalizeAlertmanagerValue,
+  },
+};
+
+const Status = () => {
+  const { response: data, error, isLoading } = useFetch<StatusPageState[]>(ENDPOINTS);
```

Comment:
You could consider partially rendering the page when each response comes back. Since each section of the page seems to be built on a corresponding api call, it might work nicely, be easy to implement, and render quicker.

Reply:
Currently, the page is loading for a few ms and I guess adding spinners for each section can lead to bad user experience. But even if it takes more I still prefer to see one spinner. Also managing multiple requests separately will lead to more complexity in the code. It's just a matter of what the requirements are. What do YT? @csmarchbanks  @juliusv

Reply:
So I really wasn't sure about this, so I tried out a version locally where it loads every section separately with its own spinner. In the end I think I prefer having the single load / spinner vs. multiple ones, since the requests are really fast and the individual loading that appears but disappears almost too fast to see looks less good than the full-page load.